### PR TITLE
feature(zms): add new properties in Context

### PR DESCRIPTION
**Pull request by hiha3456**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/framework/context.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import numpy as np
 import dataclasses
+import torch
```
Comment (PaParaZz1) [inline ding/framework/context.py:3]: why import torch here

File: ding/framework/context.py [hunk 2]
```diff
@@ -49,8 +50,12 @@ class OnlineRLContext(Context):
     env_episode: int = 0
     train_iter: int = 0
     train_data: Union[Dict, List] = None
+    train_output: Union[Dict, List] = None
```
Comment (PaParaZz1) [inline ding/framework/context.py:53]: Dict -> List[Dict]

File: ding/framework/context.py [hunk 3]
```diff
@@ -49,8 +50,12 @@ class OnlineRLContext(Context):
...
     # collect
     collect_kwargs: Dict = dataclasses.field(default_factory=dict)
+    obs: torch.Tensor = None
```
Comment (PaParaZz1) [inline ding/framework/context.py:56]: maybe `Union[torch.Tensor, Dict[torch.Tensor]]`

File: ding/framework/tests/context_fake_data.py [hunk 4]
```diff
@@ -35,6 +38,19 @@ def fake_online_rl_context():
...
+        train_output=[{
+            'cur_lr': 0.001,
+            'total_loss': random.uniform(0, 2)
+        } for _ in range(update_per_collect)],
+        obs=torch.randn(collector_env_num, obs_dim),
+        action=[np.random.randint(low=0, high=1, size=(action_dim), dtype=np.int64) for _ in range(collector_env_num)],
+        inference_output={
+            env_id: {
+                'logit': torch.randn(logit_dim),
+                'action': torch.randint(0, 1, size=(action_dim, ))
```
Comment (PaParaZz1) [inline ding/framework/tests/context_fake_data.py:50]: torch.randint(0, 1) can only generate 0

File: ding/framework/middleware/functional/collector.py [hunk 5]
```diff
@@ -61,10 +61,10 @@ def inferencer(cfg: EasyDict, policy: Policy, env: BaseEnvManager) -> Callable:
     def _inference(ctx: "OnlineRLContext"):
         """
         Output of ctx:
-            - obs (:obj:`Dict[Tensor]`): The input states fed into the model.
+            - obs (:obj:`torch.Tensor`): The input observations collected from all collector environments.
```
Comment (PaParaZz1) [inline ding/framework/middleware/functional/collector.py:64]: modify here like ctx

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: ding/framework/context.py (hunk 2 above)
Comment (hiha3456) [reply ding/framework/context.py:53]: train_output has two cases, in trainer it is Dict, in learner it is List[Dict]. Should we only define the case of learner here?

File: ding/framework/context.py (hunk 1 above)
Comment (hiha3456) [reply ding/framework/context.py:3]: Because in line 56 I used torch.Tensor for typing

Comment (PaParaZz1) [reply ding/framework/context.py:3]: use `TYPE_CHECKING` flag like [this](https://github.com/opendilab/DI-engine/blob/main/ding/framework/middleware/functional/ctx_helper.py#L6)

File: ding/framework/context.py (hunk 2 above)
Comment (PaParaZz1) [reply ding/framework/context.py:53]: Union[Dict, List[Dict]]

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**